networking/eth2/src/main/java/tech/pegasys/artemis/networking/eth2/gossip/topics/validation/AttestationValidator.java

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.networking.eth2.gossip.topics.validation;
+
+import static com.google.common.primitives.UnsignedLong.ONE;
+import static com.google.common.primitives.UnsignedLong.ZERO;
+import static tech.pegasys.artemis.datastructures.util.AttestationUtil.get_indexed_attestation;
+import static tech.pegasys.artemis.datastructures.util.AttestationUtil.is_valid_indexed_attestation;
+import static tech.pegasys.artemis.networking.eth2.gossip.topics.validation.ValidationResult.INVALID;
+import static tech.pegasys.artemis.networking.eth2.gossip.topics.validation.ValidationResult.SAVED_FOR_FUTURE;
+import static tech.pegasys.artemis.networking.eth2.gossip.topics.validation.ValidationResult.VALID;
+import static tech.pegasys.artemis.util.config.Constants.ATTESTATION_PROPAGATION_SLOT_RANGE;
+import static tech.pegasys.artemis.util.config.Constants.SECONDS_PER_SLOT;
+import static tech.pegasys.artemis.util.config.Constants.VALID_ATTESTATION_SET_SIZE;
+
+import com.google.common.primitives.UnsignedLong;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import tech.pegasys.artemis.datastructures.operations.Attestation;
+import tech.pegasys.artemis.datastructures.operations.IndexedAttestation;
+import tech.pegasys.artemis.datastructures.state.BeaconState;
+import tech.pegasys.artemis.datastructures.util.CommitteeUtil;
+import tech.pegasys.artemis.storage.client.RecentChainData;
+import tech.pegasys.artemis.util.collections.ConcurrentLimitedSet;
+import tech.pegasys.artemis.util.collections.LimitStrategy;
+import tech.pegasys.artemis.util.config.Constants;
+
+public class AttestationValidator {
+
+  private static final UnsignedLong MILLIS_PER_SECOND = UnsignedLong.valueOf(10000);
```

Comment:
Isn't this supposed to be:
```suggestion
  private static final UnsignedLong MILLIS_PER_SECOND = UnsignedLong.valueOf(1000);
```

Reply:
True.  Turns out it doesn't matter much as both current time and attestation time are in seconds anyway. Both become equally bigger and the tolerance is still less than one second.